### ISSUE TITLE
Empty DATE tags seem to cause a crash

### ISSUE THREAD

**Issue by ThierryS75** [andy-z/ged4py#5]
* GEDCOM parser for Python version: 0.1.9
* Python version: 3.7
* Operating System: Windows 10

### Description

Trying to parse a GEDCOM file and iterate over INDI records.

### What I Did
```
from ged4py import GedcomReader
parser = GedcomReader("C:/Users/Thierry/PycharmProjects/Learning/XXX.ged")
for record in parser.records0("INDI"):
    print("Level=", record.level, "  xref_id=", record.xref_id, "  tag=", record.tag, "  name=", record.name, sep="")
```

### Traceback
```
Traceback (most recent call last):
  File "C:/Users/Thierry/PycharmProjects/Learning/Ged4py Learn.py", line 5, in <module>
    for record in parser.records0("INDI"):
  File "C:\Users\Thierry\PycharmProjects\Learning\venv\lib\site-packages\ged4py\parser.py", line 320, in records0
    yield self.read_record(offset)
  File "C:\Users\Thierry\PycharmProjects\Learning\venv\lib\site-packages\ged4py\parser.py", line 358, in read_record
    rec.freeze()
  File "C:\Users\Thierry\PycharmProjects\Learning\venv\lib\site-packages\ged4py\model.py", line 349, in freeze
    self.value = DateValue.parse(self.value)
  File "C:\Users\Thierry\PycharmProjects\Learning\venv\lib\site-packages\ged4py\detail\date.py", line 221, in parse
    m = regex.match(datestr)
TypeError: expected string or bytes-like object

Process finished with exit code 1
```
This seems to be caused by an empty DATE tag?

Many thanks,

Thierry

**Comment by andy-slac** [comment]
Thanks for bug report. The DATE tag cannot be empty, at least not according to GEDCOM standard. What application did you use to produce a GEDCOM file? Some applications take a liberal approach in following the standards, it's hard to support all possible inconsistencies. If you can share your GEDCOM file (you can send it tom my email if you are not comfortable attaching it here) I could try to add some specific workarounds. Pull requests are also welcome.

**Comment by andy-z** [comment]
Thanks for bug report. The DATE tag cannot be empty, at least not according to GEDCOM standard. What application did you use to produce a GEDCOM file? Some applications take a liberal approach in following the standards, it's hard to support all possible inconsistencies. If you can share your GEDCOM file (you can send it tom my email if you are not comfortable attaching it here) I could try to add some specific workarounds. Pull requests are also welcome.

**Comment by andy-slac** [comment]
Thanks for an example file. I'll try to add a workaround for that. You can also try to file a bug report for Ancestris, I believe it is 100% against standard to have DATE tag without any value (and it does not make any sense to have that)

**Comment by andy-z** [comment]
Thierry, I have started working on Ancestris support, one other potential issue is the name format. GEDCOM standard is not very specific in that area so different apps follow different conventions for encoding names, especially maiden or married names. I have no time to install Ancestris myself and investigate its behavior. What could help me is few examples from your large file which shows how maiden names or married names appear here. In the file that you uploaded there is one example:

    1 NAME Thierry/S/
    2 GIVN Thierry
    2 SURN S

If you could copy few more examples like this from your large file for persons which also have maiden or married names which are different that would be very helpful.

You can also check a discussion https://ged4py.readthedocs.io/en/latest/techinfo.html#name-representation if you are interested in details for applications that I know about.

**Comment by ThierryS75** [comment]
Hi Andy,

I've attached two Ancestris example files, as well as an extended personal sample file, with one entry where I've completed every name-related tag Ancestris allows me to. I don't use married names in my own .ged files, and Ancestris doesn't allow me to specify them in its two user-friendly editors ("Cygnus" and "Aries") - but it lets me add a second name tag using the Gedcom editor, and I can also add a TYPE subtag to that. I've included an example of this.

[bourbon.ged.txt](https://github.com/andy-z/ged4py/files/2481599/bourbon.ged.txt)
[kennedy.ged.txt](https://github.com/andy-z/ged4py/files/2481600/kennedy.ged.txt)
[Test-ged4py.ged.txt](https://github.com/andy-z/ged4py/files/2481663/Test-ged4py.ged.txt)

I will talk to the lead Ancestris developer about the empty DATE issue.

Many thanks,

Thierry
